### ISSUE TITLE
Explore home page lists applications as snap only despite there being deb versions

### ISSUE THREAD

**Issue by local-optimum** [ubuntu-flutter-community/software#867]
![image](https://user-images.githubusercontent.com/55029393/214309221-ea4b2f6f-78d2-4f8d-960a-5375f10898f1.png)

The default screen for the store shows all featured applications as snaps only. However there are deb versions available for a number of them, in the above screenshot you can see Kdenlive displays only the snap logo. However when you search for it the deb and snap icons are visible.

Expected behaviour: Deb and Snap icons are always visible in all circumstances for applications that are available as both a deb and a snap.

![image](https://user-images.githubusercontent.com/55029393/214310073-8c9f99c4-f238-4bdc-a55b-573c3e8daf59.png)

**Comment by Feichtmeier** [comment]
Yep, we need a solution for this that does not slow down the start up even more. Those snaps you see there are related to the snap category "featured" and on the front page currently this is all related to the snap categories coming directly from the store. The easiest would be if the store would show us the corresponding snap as a field, but if this is not possible we would need to search for every banner you see there, again, for appstream. This will def. slow down the start up

**Comment by local-optimum** [comment]
I don't think that would solve the problem, in every other view we show the different package formats, I think we should aim to be consistent in presentation throughout the app rather than creating a new bespoke look for this particular view.

In addition, it's not just the visual presentation, when I click on Kdenlive from the explore page it shows only the snap option, if I search for it it shows the snap and the deb in the kdenlive app page, so the confusion for the user continues even past the first impression.

![image](https://user-images.githubusercontent.com/55029393/214315515-ec2bbd48-f14b-473a-ae79-28b8bd0513b4.png)

![image](https://user-images.githubusercontent.com/55029393/214315444-2705293f-fbff-493c-bfd8-5be307c9173e.png)

**Comment by Feichtmeier** [comment]
> I don't think that would solve the problem, in every other view we show the different package formats, I think we should aim to be consistent in presentation throughout the app rather than creating a new bespoke look for this particular view.
> 
> In addition, it's not just the visual presentation, when I click on Kdenlive from the explore page it shows only the snap option, if I search for it it shows the snap and the deb in the kdenlive app page, so the confusion for the user continues even past the first impression.
> 
> ![image](https://user-images.githubusercontent.com/55029393/214315515-ec2bbd48-f14b-473a-ae79-28b8bd0513b4.png)
> 
> ![image](https://user-images.githubusercontent.com/55029393/214315444-2705293f-fbff-493c-bfd8-5be307c9173e.png)
> 

The corresponding result from the snap only banner is a snap only app page you've shown. Fixing the snap deb mapping for the start page would automatically fix the other issue you've shown.

If someone comes up with a solution that doesn't slow down the start page I'm all in 👍

**Comment by Feichtmeier** [comment]
on the topic of the deadline:

sorry for going a bit off-topic but: my personal opinion and feature goals set for this app have been reached and IMO we have all the features we need plus much improved performance, design, UX and features over the gnome software version. This one is non-critical in my opinion. We need to take really care that we do not reduce the performance of this application. This here is a performance risk. I will mark requests from now on as "might impact performance" that do so, to make it transparent for you all

**Comment by robert-ancell** [comment]
In previous work I've done on software UIs I think the solution to startup problems is to use a cache for some of the package information. This means you can have almost instant startup, and then update the UI later if it has changed from the cached version. In most cases, this is very unlikely and will only show visually in a small UI change (e.g. a summary text changing).

Because the number of packages that the user interacts with will be relatively small the cache shouldn't use any significant disk or memory as long as older entries are dropped.

**Comment by d-loose** [comment]
#1006 adds the deb versions to the start page apps as soon as appstream is available.

Still unresolved: if you click on an app before appstream finishes loading, the app page you land on won't be updated if a debian package is available until you navigate back and forth again. This will need a bit more refactoring :)
